src/Workspaces/Core/Portable/Shared/Extensions/LocationExtensions.cs

```diff
@@ -9,16 +9,16 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions
     internal static class LocationExtensions
     {
         public static SyntaxToken FindToken(this Location location, CancellationToken cancellationToken)
-            => location.SourceTree.GetRoot(cancellationToken).FindToken(location.SourceSpan.Start);
+            => location.SourceTree!.GetRoot(cancellationToken).FindToken(location.SourceSpan.Start);
```

Comment:
@jasonmalinowski thoughts on this one?

Reply:
I'd add a Contract.ThrowIfNull() for location.SourceTree, so if somebody does pass in a location without a tree the failure is clear versus the other possible null refs that you might hit.

Reply:
That amounts to a major refactoring of the code base as it's quite common for the code `location.SourceTree != null` to exist (as well as the opposite). Happy to file a follow up issue to look into fixing this but want to keep refactoring separate from annotations

---
In reply to: [356892086](https://github.com/dotnet/roslyn/pull/40268#discussion_r356892086) [](ancestors = 356892086)

Reply:
@jaredpar I believe Jason means adding a `ThrowIfNull` here inside `FindToken`, so we get a more reasonable exception than `NullReferenceException` if an unsupported `Location` is used with this extension method.

Reply:
How about instead we add `GetSourceTreeOrThrow` to `Location`? Implementation has the behavior that you're specifying. That keeps consumption as a simple expression in all cases we hit this. Imaging there is going to be more as we continue to annotate.

---
In reply to: [356929680](https://github.com/dotnet/roslyn/pull/40268#discussion_r356929680) [](ancestors = 356929680)